dali/sasha/saxref.cpp

```diff
@@ -1086,11 +1124,119 @@ class CNewXRefManagerBase
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch (IException *e)
+        {
+            StringBuffer errMsg;
+            OWARNLOG(LOGPFX "Exception getting Sasha plane: %s, falling back to Dali storage", e->errorMessage(errMsg).str());
+            e->Release();
+            useSashaPlane = false;
+        }
+        
+        if (useSashaPlane)
+        {
+            // Save branches to Sasha plane files
+            bool saveSuccess = true;
+            saveSuccess = saveBranchToSashaPlane(sashaDir.str(), "Orphans", orphansbranch) && saveSuccess;
+            saveSuccess = saveBranchToSashaPlane(sashaDir.str(), "Lost", lostbranch) && saveSuccess;
+            saveSuccess = saveBranchToSashaPlane(sashaDir.str(), "Found", foundbranch) && saveSuccess;
+            saveSuccess = saveBranchToSashaPlane(sashaDir.str(), "Directories", dirbranch) && saveSuccess;
+            
+            // Save Messages
+            Owned<IPropertyTree> message = createPTree("Messages");
+            ForEachItemIn(i1,errors) {
+                cMessage &item = errors.item(i1);
+                IPropertyTree *t = message->addPropTree("Error",createPTree("Error"));
+                t->addProp("File",item.lname.get());
+                t->addProp("Text",item.msg.get());
+            }
+            ForEachItemIn(i2,warnings) {
+                cMessage &item = warnings.item(i2);
+                IPropertyTree *t = message->addPropTree("Warning",createPTree("Warning"));
+                t->addProp("File",item.lname.get());
+                t->addProp("Text",item.msg.get());
+            }
+            saveSuccess = saveBranchToSashaPlane(sashaDir.str(), "Messages", message) && saveSuccess;
+            
+            if (saveSuccess)
+            {
+                // Store path reference in Dali instead of full data
+                // Convert to hostname-based URL if not a local path
+                StringBuffer pathUrl;
+                if (!isAbsolutePath(sashaDir.str()))
+                {
+                    // If not absolute, make it a URL with hostname
+                    StringBuffer hostname;
+                    queryHostIP().getHostText(hostname);
+                    pathUrl.append("file://").append(hostname).append(sashaDir);
+                }
+                else
+                {
+                    pathUrl.append(sashaDir);
+                }
+                croot->setProp("@xrefPath", pathUrl.str());
+                PROGLOG(LOGPFX "Saved XREF data to Sasha plane with path reference: %s", pathUrl.str());
+            }
+            else
+            {
+                OWARNLOG(LOGPFX "Failed to save some branches to Sasha plane, falling back to Dali storage");
+                useSashaPlane = false;
+            }
+        }
+        
+        if (!useSashaPlane)
```

Comment:
given we are not going to failover to this. Just make this the else branch of if (useSashaPlane)